src/Nethermind/Nethermind.BeaconNode/MockedStart/QuickStart.cs

```diff
@@ -224,20 +208,22 @@ private byte[] GeneratePrivateKey(ulong index)
         {
             var input = new Span<byte>(new byte[32]);
             var bigIndex = new BigInteger(index);
-            var success = bigIndex.TryWriteBytes(input, out var bytesWritten);
-            if (!success || bytesWritten == 0)
+            var indexWriteSuccess = bigIndex.TryWriteBytes(input, out var indexBytesWritten);
+            if (!indexWriteSuccess || indexBytesWritten == 0)
             {
                 throw new Exception("Error getting input for quick start private key generation.");
             }
 
-            var hash = _cryptographyService.Hash(input).AsSpan();
-
-            var value = new BigInteger(hash.ToArray());
+            var hash32 = _cryptographyService.Hash(input);
+            var hash = hash32.AsSpan();
+            // Mocked start interop specifies to convert the hash as little endian (which is the default for BigInteger)
+            var value = new BigInteger(hash.ToArray(), isUnsigned: true);
```

Comment:
use UInt256 instead of BigInteger everywhere

Reply:
I didn't know what Dirichlet was and my first response was going to be simplicity of using .NET Core Numerics rather than a 3rd party.  But, had a quick look first, and Dirichlet is not a bignumber replacement, but specifically targets Int128/UInt128, and your extensions for Int256/UInt256.

The other consideration is that this code is only called at initial startup, and only when running in test network mode; it is mock code for test networks and not applicable for production mainnet, so performance is not a concern.

I actually note that because it has a crazy way (the code I copied from the test specs) that re-calculates the Merkle tree for the deposit proof for each deposit (it could just cache results and only change the latest).

But, consistency is important, so easy enough to change.